stitches.config.ts

```diff
@@ -0,0 +1,23 @@
+import { createStitches } from '@stitches/react';
+import { colorPrimitives } from './src/primitives/colorPrimitives';
+
+export const {
+  styled,
+  css,
+  globalCss,
+  keyframes,
+  getCssText,
+  theme,
+  createTheme,
+  config,
+} = createStitches({
+  theme: {
+    colors: colorPrimitives,
+  },
+  media: {
+    bp1: '(min-width: 480px)',
+  },
+  utils: {
+    marginX: (value: string) => ({ marginLeft: value, marginRight: value }),
```

Comment:
We might want to type `value` a bit more strictly than `string`. One thing that stitches provides is a `spacing` configuration -- here's mine from another project.

```
space: {
    1: '8px',
    2: '16px',
    3: '24px',
    4: '32px',
    half: '4px',
    onePlusHalf: '12px',
  },
```

(you'll want to investigate what spacing tokens are used in the UI kit and make your own config)

This allows you to restrict the values that you pass to `marginX` to the available spacing by using Template Literal Types:

https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html

If you type `value` as `value: `$${keyof typeof theme['space']}` | (string & {})`, it will allow you to use `string & {}` as an escape hatch, but you'll also get TS autocomplete and checking for `keyof typeof theme['space']}`. Let me know if you have any questions on this one!

Reply:
Thanks @Boeing787, this is super useful! I've now added the space system from the Figma into the config 👌 

I had no idea that you could do that with the type, very cool. I get the approach and attempted to implement verbatim but it hit some issues. I'll give this another go tomorrow depending on what you think of what is below. I've got a lot to learn with typescript still!

I've just come across the stitches TypeScript features: https://stitches.dev/docs/typescript
So we could write this util like this: 
`mx: (value: Stitches.ScaleValue<'space'>) => ({ marginLeft: value, marginRight: value })`
Does this do the same thing?

Reply:
@Boeing787 Are you happy for me to roll with `Stitches.ScaleValue<'space'>` for the type declaration here and to resolve this convo?